livingatlas/pipelines/src/main/java/au/org/ala/pipelines/converters/ALAOccurrenceJsonConverter.java

```diff
@@ -0,0 +1,425 @@
+package au.org.ala.pipelines.converters;
+
+import static org.gbif.pipelines.core.utils.ModelUtils.extractOptValue;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.Builder;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.gbif.dwc.terms.DwcTerm;
+import org.gbif.dwc.terms.GbifTerm;
+import org.gbif.pipelines.core.converters.JsonConverter;
+import org.gbif.pipelines.io.avro.*;
+import org.gbif.pipelines.io.avro.json.*;
+
```

Comment:
Needs a head comment. This is complicated enough that some description of what is going on should help the innocent.

Reply:
comments added